javascript/ql/src/Security/Summaries/Shared.qll

```diff
@@ -0,0 +1,95 @@
+/**
+ * Provides utility predicates for working with flow summary specifications.
+ */
+
+import javascript
+
+/**
+ * Holds if `config` matches `spec`, that is, either `spec` is the name of `config`
+ * or `spec` is the empty string and `config` is an arbitrary configuration.
+ */
+predicate configSpec(DataFlow::Configuration config, string spec) {
+  config.toString() = spec
+  or
+  spec = "" and
+  not config.isBarrier(_) and
+  not config.isBarrier(_, _)
```

Comment:
There seems to be negative recursion going through here.

The characteristic predicate of the dataflow configuration in `ImportFromCsv` depends on `configSpec` which depends on the dispatch predicate of `isBarrier`, which depends on the set of dataflow configurations.

Reply:
Oh dear, I wonder when that crept in. I'll take a look on Monday.

Reply:
Fixed by removing the two negative conjuncts. They weren't a good idea in the first place.